### PR TITLE
[DOCS] Adds sync to data frame transform API

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds the sync property to the create data frame transform API reference (https://www.elastic.co/guide/en/elasticsearch/reference/master/put-data-frame-transform.html)

### CODE REVIEW

File: docs/reference/data-frames/apis/put-transform.asciidoc [hunk 1]
```diff
@@ -67,6 +67,15 @@ IMPORTANT:  You must use {kib} or this API to create a {dataframe-transform}.
 `source`::
   (object) Required. The source configuration, which consists of `index` and 
   optionally a `query`. See <<data-frame-transform-source>>.
+  
+`sync`::
+  (Optional, object) Defines the properties required to run continuously.
+  `field`:::
```
Comment (hendrikmuhs) [inline docs/reference/data-frames/apis/put-transform.asciidoc]: there is another level between sync and field/delay, its "sync" -> "time" -> "field"/"delay" (see your example below)

sync defines continuous mode, time defines that the sync method is using a timestamp field in the data. In future we will have other ways to sync source and dest, e.g. for non timeseries data

File: docs/reference/data-frames/apis/put-transform.asciidoc [hunk 2]
```diff
@@ -67,6 +67,15 @@ IMPORTANT:  You must use {kib} or this API to create a {dataframe-transform}.
...
+    (Required, string) The date field that is used to identify new documents in
+    the source.
+  `delay`:::
+    (Optional, time units) The time delay between the current time and the
```
Comment (hendrikmuhs) [inline docs/reference/data-frames/apis/put-transform.asciidoc]: we identified yesterday that we need a default value for delay, I will update this comment once we agreed on a value.

File: docs/reference/data-frames/apis/put-transform.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/data-frames/apis/put-transform.asciidoc]: Thanks, I've drafted a description of that time property.

**Comment by sophiec20** [comment]
Adding prev discussion notes for visibility

`Please highlight that we recommend that you select a field that represents time of ingest for your time field.  
This is because we use the field to check which entities have changed since the last time we checked. If you have an ingest time field, then we can easily find what has changed.
If you use a different time field e.g. time of event, then you may need to make sure delay is set to a large enough value to allow for any delays due to the elapsed time it takes for data to reach the system...
This is a link to an example for using an ingest pipeline to set an ingest timestamp https://www.elastic.co/guide/en/elasticsearch/reference/7.2/accessing-data-in-pipelines.html#accessing-ingest-metadata` 

The `delay` default is `60s`.

**Comment by lcawl** [comment]
Thanks for the feedback @hendrikmuhs and @sophiec20 
You can preview these changes here: https://doc-review-63bf6.firebaseapp.com/es2/put-data-frame-transform.html

**Review by sophiec20 — APPROVED** [review]
LGTM

**Pull request merged**